app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload # to get the statuses into the right order (since sorted :desc)
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
@oliver-fisher you're right I'm overthinking. Let's do a very simple implementation. `emit if previous_status != new_status`. How's that?